### ISSUE TITLE
Create proper Pattern validator

### ISSUE THREAD

**Issue by declension** [quodlibet/quodlibet#1983]
#### Background

For some time, QL Pattern validation, used in various plugins worked using the approximation that a pattern could be roughtly validated by the Query validator (`Query.validator`) - but this really doesn't work any more.

#### Current impact
* Telepathy plugin uses this to validate status text patterns
* Data Editors, used extensively / exclusively by Custom Commands plugin, uses this as the central method of interpolating song data into the command argument string(s).
* The forthcoming [MQTT plugin](1976)

All the above now do no visible validation (everything is interpreted as text, rather than red or green to help the user see if they have made an obvious syntax error).

#### Validity
It's tricky to decide what a valid pattern really is, and we probably _don't_ want to change the default behaviour too much. 
* `Plain text` is valid
* `<<artist` is fairly obviously not valid
* `<~foobar11>` an unknown internal tag, is probably not valid either
* `Foo bar <||||> baz` has incorrect ternary syntax.
* `Simple > complex` is harder to say.

#### Proposal
Best bet might be to add a new specific best-efforts Pattern validator that complies with `Query.validator` (possibly via a base class), and expose a way in the parser / compiler to _not_ swallow all these exceptions if required (e.g. by this). Then move the various plugins to use this

**Comment by lazka** [comment]
Sounds good. I'm a bit confused by what Query.validator has to do with patterns..

**Comment by declension** [comment]
Yep - that's kinda the mistake (mostly my fault from ages ago I suspect) 

e.g. in [scrobbler plugin](https://github.com/quodlibet/quodlibet/blob/ef8cb9acdc51fc2172578bf802f1ee8309966849/quodlibet/quodlibet/ext/events/qlscrobbler.py#L566) or [JSONBasedEditor](https://github.com/quodlibet/quodlibet/blob/ef8cb9acdc51fc2172578bf802f1ee8309966849/quodlibet/quodlibet/qltk/data_editors.py#L142) (for Custom Commands)

**Comment by lazka** [comment]
Ah ok.

Afaik the pattern parser is quite forgiving in error cases atm because it used to be that way before the last rewrite and not all callers do proper error handling.
